### ISSUE TITLE
Delayed notifications due to connection interrupts

### ISSUE THREAD

**Issue by iRave** [microg/android_packages_apps_GmsCore#344]
I'm running Android 7.1.1 with microG core 0.2.4.79 built from source and baked directly into my ROM. The self check does not yield any error and I tested the GCM cloud with Push Notification Tester. It all seems to run fine as long as the phone stays active, however whenever doze kicks in notifications (specifically from WhatsApp) get delayed. BTW I also disabled battery optimization for microG. Whenever I would go into microG settings and check the connection timer it would always be less than an hour.
I did not know how to record a helpful logcat, so I may only submit a log from almost the whole day. Hope it helps: http://pastebin.com/DBd2XzWT

Interestingly from 1pm to 8 pm there's no message from microG in the log at all.

This might be related to #226.

**Comment by fabian-rump** [comment]
I can confirm this issue. Running Lineage 14.1 (Android 7.1.1) everything works fine for some time, but then WhatsApp messages get delayed and also the WhatsApp web service disconnects.

**Comment by ale5000-git** [comment]
@mar-v-in

This may be related:
> Apps targeting N must use Context.registerReceiver() to receive CONNECTIVITY_ACTION, declaring the intent in the AndroidManifest.xml is no longer enough. Add a service to listen for network state changes.

Ref: https://review.lineageos.org/#/c/166005/

**Comment by mar-v-in** [comment]
@ale5000-git is already done since 97f4c82172a03011e6198b652162827ba6d24da7 [here](https://github.com/microg/android_packages_apps_GmsCore/blob/master/play-services-core/src/main/java/org/microg/gms/gcm/TriggerReceiver.java#L42)

**Comment by mar-v-in** [comment]
@iRave 
Your log is very interesting:
1. The reason for notifications being no longer delivered since around 12:48 is a system service crash. This means that some component of the operating system crashed ( != microG) causing the system to try to recover from this crash. This recovery apparently does not include the recurring wake-ups of microG, so microG is no longer able to keep the link to Google servers.
2. The crash is probably caused by substratum, because substratum crashes at 12:46 and other apps follow step-by-step (search the log for "died") until the whole system is down.
3. substratum seems to use Google Firebase API, which is not fully implemented yet in microG and seems to fail initializing. This might cause the issue here, but also might be totally unrelated.

Can you please try if the problem persists with substratum removed from the system?

**Comment by forohfor** [comment]
@mar-v-in I have this problem too and I don't run substratum, once doze kicks I get no notifications despite the app being not battery optimized until I reopen the screen again.

I'm on LOS 14.1 and v0.2.4-103

**Comment by ale5000-git** [comment]
@forohfor: Hi, it would be nice if you post a full logcat as attachment.

**Comment by forohfor** [comment]
@ale5000-git [here](https://goo.gl/m9KvhP)

last time was around 3:30

**Comment by forohfor** [comment]
@ale5000-git is there a certain app causing this?

**Comment by ale5000-git** [comment]
About this: `Unable to start service Intent { act=com.google.android.gms.measurement.START pkg=com.google.android.gms } U=0: not found`
if I understand correctly NovaLauncher try to run com.google.android.gms.measurement.START that is missing.
The fact that is missing is probably because it isn't yet implemented in microG but apparently NovaLauncher try to run it again and again indefinitely and never ever give up probably using a lot of battery.
It may be a good idea to ask to developers of NovaLauncher to give up if it is missing.


Instead these are probably some problems related to your ROM that you probably should report to the ROM author:
```
Service com.android.email.service.EmailBroadcastProcessorService has leaked ServiceConnection com.android.emailcommon.service.ServiceProxy$ProxyConnection@fac846f that was originally bound here

Calling a method in the system process without a qualified user: android.app.ContextImpl.startService:1357 android.content.ContextWrapper.startService:613 android.content.ContextWrapper.startService:613 com.android.keychain.KeyChainBroadcastReceiver.onReceive:12 android.app.ActivityThread.handleReceiver:3040 

E/KernelUidCpuTimeReader(2173): Failed to read uid_cputime: /proc/uid_cputime/show_uid_stat (No such file or directory)
```

**Comment by clyso** [comment]
Having the same problem as @forohfor 
using microg as [nanomod](https://gitlab.com/Nanolx/NanoMod) magisk module on top of LOS 14.1 sultanxda edition.
Tested with Signal and Whatsapp - same behavior on both Messengers:
Once doze kicks I get no notifications despite the app being not battery optimized until I reopen the screen/app again.
Is there something specific i can look for in logcat?
Any help appreciated. Otherwise microg works really well and is a great piece of software helping to free Android 👍

**Comment by ale5000-git** [comment]
The issue of @iRave is the one explained by @mar-v-in but the issue of all other peoples is probably the one exdplained here: #338

**Comment by ale5000-git** [comment]
This should be fixed with #732.

Please try the new version released yesterday.
